src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -298,7 +298,12 @@ sl_status_t ScanCallback(sl_wifi_event_t event, sl_wifi_scan_result_t * scan_res
     sl_status_t status = SL_STATUS_OK;
     if (SL_WIFI_CHECK_IF_EVENT_FAILED(event))
     {
-        ChipLogError(DeviceLayer, "Scan Netwrok Failed: 0x%lx", *reinterpret_cast<sl_status_t *>(status));
+        if (scan_result != nullptr)
+        {
+            sl_status_t callback_status = *reinterpret_cast<sl_status_t *>(scan_result);
```

Comment:
Why are we not setting the `status` variable that is returned to the scan_results?